atomic_reactor/plugins/pre_inject_parent_image.py

```diff
@@ -0,0 +1,110 @@
+    """
+    Modifies parent image to be used based on given Koji build.
+
+    For each listed archive in Koji build, chose the archive that
+    defines a non-empty '.extra.docker.repositories' list.
+    This list provides the pull reference for the container image
+    associated with Koji build. If it contains multiple item, the
+    manifest digest, @sha256, is preferred. Otherwise, the first
+    repository in list is used.
+
+    The namespace and repository for the new parent image must match
+    the namespace and repostirory for the parent image defined in
+    Dockerfile.
+
+    This plugin returns the identifier of the Koji build used.
+    """
+
+    key = 'inject_parent_image'
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, koji_build, koji_hub, koji_ssl_certs_dir=None):
+        """
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param koji_build: str, either Koji build ID or Koji build NVR
+        :param koji_hub: str, koji hub (xmlrpc)
+        :param koji_ssl_certs_dir: str, path to "cert", "ca", and "serverca"
+                                   used when Koji's identity certificate is not trusted
+        """
+        super(InjectParentImage, self).__init__(tasker, workflow)
+
+        koji_auth_info = None
+        if koji_ssl_certs_dir:
+            koji_auth_info = {
+                'ssl_certs_dir': koji_ssl_certs_dir,
+            }
+        self.koji_session = create_koji_session(koji_hub, koji_auth_info)
+
+        self.koji_build = koji_build
+
+        self._koji_build_info = None
+        self._archive_repositories = None
+        self._new_parent_image = None
+
+    def run(self):
+        self.find_koji_build_info()
+        self.find_repositories_from_archive()
+        self.select_new_parent_image()
+        self.validate_new_parent_image()
+        self.set_new_parent_image()
+        return self._koji_build_info['id']
+
+    def find_koji_build_info(self):
+        self._koji_build_info = self.koji_session.getBuild(self.koji_build)
+        if not self._koji_build_info:
+            raise RuntimeError('Koji build, {}, not found'.format(self.koji_build))
+
+    def find_repositories_from_archive(self):
+        for archive in self.koji_session.listArchives(self._koji_build_info['id']):
+            repositories = graceful_chain_get(archive, 'extra', 'docker', 'repositories')
+            if repositories:
+                self.log.info('Using koji archive %d', archive['id'])
+                self._archive_repositories = repositories
+                break
+
+        if not self._archive_repositories:
+            raise RuntimeError('A suitable archive for Koji build {[nvr]} was not found'
+                               .format(self._koji_build_info))
+
+    def select_new_parent_image(self):
+        for repository in self._archive_repositories:
+            if '@sha256' in repository:
```

Comment:
This was implemented after I suggested in https://github.com/projectatomic/atomic-reactor/pull/770#discussion_r131769180, it probably was a bad idea. Since we can't store `self.sep` lets just revert back to original code